validation.go

```diff
@@ -67,11 +86,11 @@ func Validate(value interface{}, rules ...Rule) error {
 	switch rv.Kind() {
 	case reflect.Map:
 		if rv.Type().Elem().Implements(validatableType) {
-			return validateMap(rv)
+			return validateMap(context.Background(), rv)
```

Comment:
Is this expected? Now you are validating the map with a context, even though Validate() is being called...

Reply:
I think you raise a good point here.  Possibly the general rule should be that if you call Validate, ValidateWithContext will never be called. But if you call ValidateWithContext, it may drop down to Validate if a value / map entry etc implements Validate but not ValidateWithContext.